File: Runtime/TransitionAttributeValue.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

''' <summary>
''' Transition Attribute Value class
''' </summary>
''' <remarks></remarks>
Class TransitionAttributeValue
    Inherits AttributeValueBase

    Private m_TransitionGroupId As Integer

    Public Sub New(
        ByVal transitionAttributeTypeId As Integer,
        ByVal stratumId As Nullable(Of Integer),
        ByVal secondaryStratumId As Nullable(Of Integer),
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal transitionGroupId As Integer,
        ByVal stateClassId As Nullable(Of Integer),
        ByVal minimumAge As Nullable(Of Integer),
        ByVal maximumAge As Nullable(Of Integer),
        ByVal value As Double)

        MyBase.New(
            transitionAttributeTypeId,
            stratumId,
            secondaryStratumId,
            iteration,
            timestep,
            stateClassId,
            minimumAge,
            maximumAge,
            value)

        Me.m_TransitionGroupId = transitionGroupId

    End Sub

    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

End Class

File: Runtime/TransitionAttributeValue.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class TransitionAttributeValue : AttributeValueBase
    {
        private int m_TransitionGroupId;

        public TransitionAttributeValue(
            int transitionAttributeTypeId, int? stratumId, int? secondaryStratumId, int? tertiaryStratumId, int? iteration, int? timestep, 
            int transitionGroupId, int? stateClassId, int? minimumAge, int? maximumAge, double value) : base(transitionAttributeTypeId, stratumId, 
                secondaryStratumId, tertiaryStratumId, iteration, timestep, stateClassId, minimumAge, maximumAge, value)
        {
            this.m_TransitionGroupId = transitionGroupId;
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }
    }
}
